identity-actor/src/actor/errors.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::request_response::OutboundFailure;
+
+use crate::didcomm::ThreadId;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, thiserror::Error)]
+/// Errors that can occur during the actor operation.
+pub enum Error {
+  #[error("{context}: {source}")]
+  TransportError {
+    context: &'static str,
+    source: libp2p::TransportError<std::io::Error>,
+  },
+  #[error("invalid endpoint")]
+  InvalidEndpoint,
+  #[error("{0}")]
+  OutboundFailure(#[from] OutboundFailure),
```

Comment:
The `From` implementation brings a type from an unstable external crate into our public API. Furthermore using `#[from]` in this error implies that it is also `source` which violates the following part of our [error handling guidelines](https://github.com/iotaledger/identity.rs/issues/534):
> Return source errors via Error::source unless the source error's message is included in your own error message in Display.

Reply:
I would suggest the following here: 
1. Change `#[from]` with `#[source]` . 
2. Decorate this variant with `#[non_exhaustive]`. Together with 1. this will prevent the unstable `OutboundFailure` external type from being part of our public API. 
3. Change the error message to something that does not include the error message of the source error. 

Regarding point 2. If one needs to construct this variant from other types then wrap a `Box<dyn Error +'static + Send + Sync>` and remove the `#[non_exhaustive]` annnotation.